include/hyprtoolkit/core/Backend.hpp

```diff
@@ -73,6 +80,12 @@ namespace Hyprtoolkit {
 
         Hyprutils::Memory::CSharedPointer<CPalette> getPalette();
 
+        void                                        unlockSession();
+
+        struct {
+            Hyprutils::Signal::CSignalT<uint32_t> outputAdded;
```

Comment:
brother eeeww, don't pass an uint32_t, pass a `SP<IOutput>`

Also we need a std::vec<> getOutputs()